glean-core/ios/GleanTests/GleanTests.swift

```diff
@@ -3,22 +3,139 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 @testable import Glean
+import OHHTTPStubs
 import XCTest
 
 class GleanTests: XCTestCase {
+    var expectation: XCTestExpectation?
+
     override func setUp() {
-        // Put setup code here. This method is called before the invocation of each test method in the class.
+        Glean.shared.resetGlean(clearStores: true)
+        Glean.shared.enableTestingMode()
     }
 
     override func tearDown() {
         Glean.shared.setUploadEnabled(true)
     }
 
+    private func setupHttpResponseStub(statusCode: Int32 = 200) {
+        expectation = expectation(description: "Ping sent")
+        // This function will be handling one each of baseline, events, and metrics pings
+        // so we set the expected count to 3 and set it to assert for overfulfill in order
+        // to test that unknown pings aren't being sent.
+        expectation!.expectedFulfillmentCount = 3
+        expectation!.assertForOverFulfill = true
+        let host = URL(string: Configuration.Constants.defaultTelemetryEndpoint)!.host!
+        stub(condition: isHost(host)) { data in
+            let body = (data as NSURLRequest).ohhttpStubs_HTTPBody()
+            let json = try! JSONSerialization.jsonObject(with: body!, options: []) as? [String: Any]
+            XCTAssert(json != nil)
+            let pingType = (json?["ping_info"] as? [String: Any])!["ping_type"] as? String
+            XCTAssertTrue(Glean.shared.testHasPingType(pingType!))
+
+            self.expectation!.fulfill()
+
+            return OHHTTPStubsResponse(
+                jsonObject: [],
+                statusCode: statusCode,
+                headers: ["Content-Type": "application/json"]
+            )
+        }
+    }
+
     func testInitializeGlean() {
-        let glean = Glean.shared
+        Glean.shared.initialize()
```

Comment:
With the above `setUp` method we already initialize Glean, making this the second initialize, which shouldn't be a problem, but it will log an error and not re-initialize.

That doesn't make the test wrong, but we could also remove that line and add a comment